tests/unit/models/gpflow/test_models.py

```diff
@@ -407,6 +407,22 @@ def test_gaussian_process_regression_raises_for_invalid_init() -> None:
         GaussianProcessRegression(gpr_model(x, y), optimizer=optimizer2)
 
 
+def test_gaussian_process_regression_correctly_inits_mean_function() -> None:
+    x_np = np.arange(5, dtype=np.float64).reshape(-1, 1)
+    x = tf.convert_to_tensor(x_np, x_np.dtype)
+    y = fnc_3x_plus_10(x)
+
+    m = gpflow.models.GPR((x, y), gpflow.kernels.RBF())
+    model = GaussianProcessRegression(m)
+    assert isinstance(model.get_mean_function(), gpflow.mean_functions.Zero)
+
+    m = gpflow.models.GPR(
+        (x, y), gpflow.kernels.RBF(), mean_function=gpflow.mean_functions.Linear()
+    )
+    model = GaussianProcessRegression(m)
+    assert isinstance(model.get_mean_function(), gpflow.mean_functions.Linear)
```

Comment:
(Why not check that it's the actual mean function in the model, rather than just the same type?)